presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -103,10 +109,14 @@ private PrestoDriverUri(URI uri, Properties driverProperties)
     {
         this.uri = requireNonNull(uri, "uri is null");
         address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
-        properties = mergeConnectionProperties(uri, driverProperties);
+        properties = mergeProvidedProperties(uri, driverProperties);
+
+        setSSLPropertyByPortIfNotProvided(properties, uri);
 
         validateConnectionProperties(properties);
 
+        setDefaultPropertiesIfNotProvided(properties);
```

Comment:
If we set default values to `SSL` and `SSL_VERIFICATION`, in the following case, the previous code will throw an exception `java.sql.SQLException: Connection property 'SSLVerification' is not allowed`:

- Both of `SSL` and `SSL_VERIFICATION` are not provided in the URI or driver properties
- The default values of `SSL` and `SSL_VERIFICATION` are merged into the variable `properties`
- `SSL` is default `false`
- `SSL=false` and `SSL_VERIFICATION` in `properties` => Exception

I tried to solve this problem by first validating the provided properties, then merging the default properties into the variable `properties` without overwriting.

Reply:
Maybe `SSL_VERIFICATION` should not have default value then or be ignored when `SSL=false`?

Reply:
When `SSL=false` is set by the user and `SSL_VERIFICATION` is also set by the user. An exception should be thrown stating that `SSL_VERIFICATION` is not allowed.

If there is a default value for `SSL_VERIFICATION` and the logic to set the properties remains the same, I am not sure how to tell if the default value is overwritten or not and such information is required to determine if `SSL_VERIFICATION` is allowed or not.

For example, for the following cases: 
1. `SSL_VERIFICATION=FULL` (set by the user), and `SSL` is not set. 
2. `SSL_VERIFICATION=FULL` (set by the user), and `SSL=false` is set. 
3. `SSL_VERIFICATION=FULL` (by default), and `SSL` is not set. 
4. `SSL_VERIFICATION=FULL` (by default), and `SSL=false` is set. 

Case 1 and 2 should throw an exception.
Case 3 and 4 should ignore the default value of `SSL_VERIFICATION`  and no exception should be thrown.

@kokosing SSLVerification parameter without a default value would be easier to be implemented. Should we implement it without a default value?

Reply:
> Should we implement the SSLVerification parameter without a default value?

Yes, that sounds like a good idea. Thank you!

Reply:
Thank you for the confirmation. Will implement it without a default value.